src/printer.js

```diff
@@ -609,24 +605,45 @@ function genericPrintNoParens(path, options, print) {
     case "ReturnStatement":
       parts.push("return");
 
-      if (
-        n.argument &&
-        n.argument.comments &&
-        n.argument.comments.some(comment => comment.leading)
-      ) {
-        parts.push(
-          concat([
-            " (",
-            indent(
-              options.tabWidth,
-              concat([softline, path.call(print, "argument")])
-            ),
-            line,
-            ")"
-          ])
-        );
-      } else if (n.argument) {
-        parts.push(" ", path.call(print, "argument"));
+      if (n.argument) {
+        const comments = n.argument.comments || [];
+        if (
+          n.argument.type === "BinaryExpression" ||
+          n.argument.type === "LogicalExpression" ||
+          n.argument.type === "ConditionalExpression" ||
+          n.argument.type === "CallExpression" ||
+          n.argument.type === "MemberExpression" ||
+          n.argument.type === "SequenceExpression" ||
+          n.argument.type === "TaggedTemplateExpression"
+        ) {
+          function getLeftSide(node) {
```

Comment:
I also considered a switch/case explicitly listing the node types corresponding to each relevant node attribute; figured to go with this since it's more concise, but could have subtle bugs since only the top node's type is verified, and ~any expression could be inside its stack...